src/components/Suspense/SuspendConditionally.tsx

```diff
@@ -0,0 +1,21 @@
+import React from 'react'
+import { useState } from 'react'
+
+export const SuspendConditionally = (props: { if: boolean; children: React.ReactNode }) => {
+  useSuspendIf(props.if)
+  return <>{props.children}</>
```

Comment:
Can't this just be returned as-is?

```suggestion
  return props.children
```

Reply:
type system complained.